src/page_entry.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fs;
+use std::path::Path;
+use std::path::PathBuf;
+
+use eyre::{eyre, WrapErr};
+use url::Url;
+use walkdir::WalkDir;
+
+use crate::entry::EntryMetadata;
+use crate::entry_util::{check_mismatched_files, PathPair, METADATA_FILE_EXT, POST_FILE_EXT};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct PageEntry {
+    pub metadata: EntryMetadata,
```

Comment:
I think there is actually value in keeping the fields. Consider Wikipedia: it's all pages, but it shows when an article is last updated. Keeping track of a modified date is useful, but not all pages need it. But there's plenty of times when a site owner might want to communicate the last updated date to the reader of a page.

Reply:
Oh I agree that stuff like the last modified date has utility, but users could just define their own `updated` field in the `values` section and include it in their page template.

Ultimately all the fields in the metadata file are just being piped into the page template, so there's no need to make some of them magic unless we're actually implementing special behavior for them (I think we should keep `draft`, because it *does* have special behavior).

Another problem is that `updated` is a *required* field. That makes sense for posts, because the Atom spec requires it, but we shouldn't force users to have a last-modified time for pages if they don't want/need to.

Reply:
True. Then perhaps keeping `draft` and `values` is enough. There is some argument for `title` as well, although that could be under `values` too. But if there's even one required field, that makes the yaml required, which i would say also is an argument for keeping the `new-page` command.

Reply:
I think making the YAML file completely optional for pages is a good idea! I could see two use-cases for that:

- Users want a static header/footer on every page, and don't necessarily need to reference metadata in their pages template.
- The pages template references metadata, but conditionally, so not every page needs to define that metadata.

I really like the idea that users can just drop a `.gmi` file in a directory and have it automatically templated.

Reply:
I think a good idea might be then to provide an optional switch to the new page command that also generates a yaml file. By default it doesn't.